compiler/parser.mly

```diff
@@ -81,6 +80,8 @@ expr:
   | literal                               { $1 }
   | arith                                 { $1 }
   | boolean                               { $1 }
+  | dist                                  { Dist($1) }
+  | dist_func                             { Dist($1) }
```

Comment:
can we have both dist options be pattern matches under one `dist` branch? like we did in arith, boolean, etc